File: C#/uri-1028.cs
// URI - 1028: https://www.urionlinejudge.com.br/judge/pt/problems/view/1028

using System;

namespace URI
{
    class Program
    {
        static void Main(string[] args)
        {
            int numberOfExchanges = int.Parse(Console.ReadLine());
            string[] numberOfStickers = new string[2];
            int ricardoStickers, vicenteStickers;

            for (int i = 0; i < numberOfExchanges; i++)
            {
                numberOfStickers = Console.ReadLine().Split(' ');

                ricardoStickers = int.Parse(numberOfStickers[0]);
                vicenteStickers = int.Parse(numberOfStickers[1]);

                Console.WriteLine(GreatestCommonDivisor(ricardoStickers, vicenteStickers));
            }
        }

        private static int GreatestCommonDivisor(int ricardoStickers, int vicenteStickers)
        {
            int remainder;

            while (vicenteStickers != 0)
            {
                remainder = ricardoStickers % vicenteStickers;
                ricardoStickers = vicenteStickers;
                vicenteStickers = remainder;
            }

            return ricardoStickers;
        }
    }
}

File: Ruby/uri-1028.rb
#URI - 1028: https://www.urionlinejudge.com.br/judge/pt/problems/view/1028

number_of_test_cases = gets.to_i

number_of_test_cases.times do 
    input = gets.split(' ')
    number_of_cards_from_ricardo = input[0].to_i
    number_of_cards_from_vicente = input[1].to_i
    maximum_number_of_cards_to_be_exchanged = number_of_cards_from_ricardo.gcd(number_of_cards_from_vicente)
    puts maximum_number_of_cards_to_be_exchanged
end